nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import math
+import random
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import numpy as np
+import torch
+import torch.utils.data as pt_data
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+from pytorch_lightning.utilities import rank_zero_only
+from sacrebleu import corpus_bleu
+
+from nemo.collections.common.losses import SmoothedCrossEntropyLoss
+from nemo.collections.common.metrics import Perplexity
+from nemo.collections.common.parts import transformer_weights_init
+from nemo.collections.nlp.data import TranslationDataset
+from nemo.collections.nlp.data.machine_translation.machine_translation_dataset import TranslationDataConfig
+from nemo.collections.nlp.models.enc_dec_nlp_model import EncDecNLPModel, EncDecNLPModelConfig
+from nemo.collections.nlp.modules.common.transformer import BeamSearchSequenceGenerator
+from nemo.core.classes.common import typecheck
+from nemo.utils import logging
+
+__all__ = ['MTEncDecModel']
+
+
+@dataclass
+class MTEncDecModelConfig(EncDecNLPModelConfig):
```

Comment:
any reason you didn't move this class to mt_enc_dec_config.py ?

Reply:
Nope, moved it now.

Reply:
Done.